src/clojure/nrepl/util/completion.clj

```diff
@@ -8,8 +8,93 @@
   (:require [clojure.main])
   (:import [java.util.jar JarFile]
            [java.io File]
-           [java.lang.reflect Member]
-           [java.util.jar JarEntry]))
+           [java.lang.reflect Field Member]
+           [java.util.jar JarEntry]
+           [java.util.concurrent ConcurrentHashMap]))
+
+;; Code adapted from Compliment (https://github.com/alexander-yakushev/compliment)
+
+(defn annotate-keyword
+  [kw]
+  {:candidate kw :type :keyword})
+
+(defn all-keywords
+  []
+  (let [^Field field (.getDeclaredField clojure.lang.Keyword "table")]
```

Comment:
Compliment [memoizes](https://github.com/alexander-yakushev/compliment/blob/a112bb85b748d4a34da89ee783e7c64d8183dab7/src/compliment/sources/keywords.clj#L7) this function, but I'm not sure how that works exactly. If I create a new keyword, the memoized function seems to return it as well. Not sure why.

I've omitted memoization for now, but can add it if necessary.